### DISCUSSION TITLE
Socket.io: How to change server options on runtime

### DISCUSSION THREAD

**Discussion by Sarathmunusamy93** [socketio/socket.io#4342] [Q&A] [answered]
Hi Team,

Currently I am using Socket.io connection in my application. Below is my server socket.io code.

` const SocketOptions = {
      pingTimeout: 60000,
      maxHttpBufferSize: 1024 *XXX*XXX
      transports: [ "polling", "websocket"]
    }`
   `const io = socket.io( MY_SERVER, SocketOptions )`

Currently, I need to set "path" of the server option on runtime means dynamically. My question is below.

1. Is changing socket.io server options is valid?
2. If yes How to do it?

 I would be helpful if anyone help me to solve this. 

Thanks in advance.

**Comment by darrachequesne** [comment, marked as answer]
Hi! You can use the `path()` method:

```js
import { Server } from "socket.io";

const io = new Server(3000);

io.path("/myownpath/");
```

Reference: https://socket.io/docs/v4/server-api/#serverpathvalue

For the other [options](https://socket.io/docs/v4/server-options/), you may need to update the `opts` attribute of the underlying engine, for example:

```js
io.engine.opts.maxHttpBufferSize = 1e8;
```
